CorsixTH/Lua/hospital.lua

```diff
@@ -2274,14 +2303,12 @@ end
 
 --! Function that returns true if the room for the given disease
 --! has not been researched yet.
---! param disease (string): the disease to be checked.
+--!param disease (string): the disease to be checked.
 function Hospital:roomNotYetResearched(disease)
   local req = self:checkDiseaseRequirements(disease)
   if type(req) == "table" and #req.rooms > 0 then
     for _, room_id in ipairs(req.rooms) do
-      if not self.discovered_rooms[self.world.available_rooms[room_id]] then
-        return true
-      end
+      if not self:isRoomDiscovered(room_id) then return true end
     end
```

Comment:
As a suggestion for a next patch, negated logic (true if something is not the case) is mostly horrible for reasoning, code is much easier to understand and reason about if a positive result yields true.

Reply:
To be fair to lewri, this method already existed with that logic.